variables.tf

```diff
@@ -186,6 +186,24 @@ variable "auto_verified_attributes" {
   ]
 }
 
+variable "account_recovery_setting_recovery_mechanisms" {
+  type        = any
+  description = "(Optional) Account recovery mechanisms"
+  # Example:
+  #
+  # account_recovery_setting_recovery_mechanisms = [
+  #   {
+  #     name          = "verified_email"
+  #     priority      = 1
+  #   },{
+  #     name          = "verified_phone_number"
+  #     priority      = 2
+  #   }]
+
+  default = []
```

Comment:
what is the actual terraform or AWS default here?

Reply:
<img width="985" alt="image" src="https://user-images.githubusercontent.com/679977/109834133-2c6da100-7c42-11eb-985b-0da2a49d9c4d.png">

Reply:
that's a bad default... ;) 

the default should then be `[email_verified, phone_verified]` .... which would be a breaking change but the recommended solution? but dependent on what attributes are actually available...

maybe instead of an array we can also just map those into defined states 

`account_recovery_mechanisms = EMAIL_OTHERWISE_PHONE | PHONE_OTHERWISE_EMAIL | EMAIL_ONLY | PHONE_ONLY ..`

naming is bad sorry just an idea for brainstorming.. but wdyt about the general idea of also implementing it in a check_box like way?

we can not cover all those cases in terraform anyway or how do you set the default in terraform i can't see any option to disallow or allow phone if MFA is also enabled via phone...
or do you know how to implement all options in code?

Reply:
I generally prefer 1:1 with terraform as this keeps things as simple as possible. The module UX will get a small hit but if you look what I actually suggested, it's also not too ugly. About the UI as driving force, UI's can also change and then you have a module which simulates the 2021-01 UI ¯\_(ツ)_/¯ . Long shot, the moment Cognito adds add a third communication method, the input on UI wil change into a list.

I leave it up to you how you want it, I'm fine with all options as long as I can implement it 👍

Reply:
yes, then let's go with 1:1 could you increase minimum required version to the version where this was added to the provider? and also add the new argument to the readme?

Reply:
👍 3.19
https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3190-december-01-2020